torchvision/prototype/datasets/utils/_resource.py

```diff
@@ -141,8 +143,40 @@ def __init__(
         super().__init__(file_name=file_name or pathlib.Path(urlparse(url).path).name, **kwargs)
         self.url = url
         self.mirrors = mirrors
+        self._resolved = False
+
+    @property
+    def resolved(self) -> bool:
+        return self._resolved
+
+    def resolve(self) -> Optional[OnlineResource]:
+        redirect_url = _get_redirect_url(self.url)
+        if redirect_url == self.url:
+            self._resolved = True
+            return self
+
+        meta = {
+            attr.lstrip("_"): getattr(self, attr)
+            for attr in (
+                "file_name",
+                "sha256",
+                "_preprocess",
+                "_loader",
+            )
+        }
+
+        gdrive_id = _get_google_drive_file_id(redirect_url)
+        if gdrive_id:
+            return GDriveResource(gdrive_id, **meta)
```

Comment:
Do we have current cases where an HTTP resource redirects to a Google drive?

Reply:
Yes. It is one of the datasets not yet ported to prototype. I've hit this while doing #5154.